src/internal/converters/eml/eml.go

```diff
@@ -303,3 +304,137 @@ func FromJSON(ctx context.Context, body []byte) (string, error) {
 
 	return email.GetMessage(), nil
 }
+
+//-------------------------------------------------------------
+// Postable -> EML
+//-------------------------------------------------------------
+
+// FromJSONPostToEML converts a postable (as json) to .eml format.
+// TODO(pandeyabs): This is a stripped down copy of messageable to
+// eml conversion, it can be folded into one function by having a post
+// to messageable converter.
```

Comment:
Assuming the function names are the same between the Postable and Messageable interfaces, we could just define our won interface that includes the subset of functions that are accessed in this code. If the function names aren't the same between the graph SDK types then we can make a new type that embeds one of the graph SDK types and implements the proper function names we need. It'll just call the corresponding function on the graph SDK type. I think we did something similar for calendar events which have `GetCalendar()` instead of `GetParentFolderId()`

Reply:
Function names aren't the same. We can make a new type like you said, or convert `Postable` to `Messageable`. Either should work out. Tracking this as part of https://github.com/alcionai/corso/issues/5155